src/commands/dev/index.ts

```diff
@@ -503,7 +503,13 @@ export default class Dev extends BaseCommand {
       build_args.push('--build-arg', arg);
     }
 
-    await DockerComposeUtils.dockerCompose(['-f', compose_file, '-p', project_name, 'build', ...build_args], { stdio: 'inherit' });
+    try {
+      await DockerComposeUtils.dockerCompose(['-f', compose_file, '-p', project_name, 'build', ...build_args], { stdio: 'inherit' });
+    } catch (e: any) {
+      const pattern = new RegExp('Command failed with exit code 17: docker compose -f [-p architect build]?', 'g');
```

Comment:
This pattern isn't sufficient, if I run `architect dev -e foo` the failure is 
```
Command failed with exit code 17: docker compose -f /Users/tyler/.config/architect/local/architect/architect/docker-compose/foo.yml -p foo build
```

the `-p` argument is equivalent to the environment we pass in.

We should be able to get the exit code from `e.exitCode` and check that value instead. Is exit code 17 the only possibility / the only exit code we care about? I imagine any non-zero exit code from `docker compose build` wouldn't be something we can fix on our end, but not 100% sure of that

Reply:
Updated logic to fetch docker compose exit code `e.exitCode` instead of parsing error message.  Updated to throw error when non-zero exit code.